File: lib/layer2_core/light/e_on_surface_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnSurfaceLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl4;
      case 'standard': return cstl4;
      case 'blue': return cbl4;
      case 'green': return cgl4;
      case 'red': return crl4;
      case 'nothing': return cnol4;
      default: return cstl4;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnSurfaceLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSurfaceLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl4
            AppTheme.STANDARD -> R.color.cstl4
            AppTheme.BLUE -> R.color.cbl4
            AppTheme.GREEN -> R.color.cgl4
            AppTheme.RED -> R.color.crl4
            AppTheme.NOTHING -> R.color.cnol4
        }
    }
}
